pkg/models/dashboards_public.go

```diff
@@ -0,0 +1,43 @@
+package models
+
+var (
+	ErrPublicDashboardFailedGenerateUniqueUid = DashboardErr{
+		Reason:     "Failed to generate unique dashboard id",
+		StatusCode: 500,
+	}
+	ErrPublicDashboardNotFound = DashboardErr{
+		Reason:     "Public dashboard not found",
+		StatusCode: 404,
+		Status:     "not-found",
+	}
+	ErrPublicDashboardIdentifierNotSet = DashboardErr{
+		Reason:     "No Uid for public dashboard specified",
+		StatusCode: 400,
+	}
+)
+
+type PublicDashboardConfig struct {
+	IsPublic        bool            `json:"isPublic"`
+	PublicDashboard PublicDashboard `json:"publicDashboard"`
+}
+
+type PublicDashboard struct {
+	Uid          string `json:"uid" xorm:"uid"`
+	DashboardUid string `json:"dashboardUid" xorm:"dashboard_uid"`
+	OrgId        int64  `json:"orgId" xorm:"org_id"`
+	TimeSettings string `json:"timeSettings" xorm:"time_settings"`
```

Comment:
I am not sure I understand this string property and how it's intended to be used. in other places where we store things like this we use *simplejson.Json, I think there is also a way to define a nested go object that automatically gets's serialized to JSON.  
But maybe this is just a placeholder for now, and it's to be replaced.

Reply:
It's a placeholder. This will be proper json.